AsyncDisplayKit/Layout/ASLayout.h

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASAssert.h>
+#import <AsyncDisplayKit/ASLayoutable.h>
+
+extern CGPoint const CGPointNull;
+
+extern BOOL CGPointIsNull(CGPoint point);
+
+/** Represents a computed immutable layout tree. */
+@interface ASLayout : NSObject
+
+@property (nonatomic, readonly) id<ASLayoutable> layoutableObject;
+@property (nonatomic, readonly) CGSize size;
+/**
+ * Position in parent. Default to CGPointNull.
+ * 
+ * @discussion When being used as a sublayout, this property must not equal CGPointNull.
+ */
+@property (nonatomic, readwrite) CGPoint position;
+/** 
+ * Array of ASLayouts. Each must have a valid non-null position.
+ */
+@property (nonatomic, readonly) NSArray *sublayouts;
+
++ (instancetype)newWithLayoutableObject:(id<ASLayoutable>)layoutableObject
+                                   size:(CGSize)size
+                               position:(CGPoint)position
+                             sublayouts:(NSArray *)sublayouts;
+
+/**
+ * Convenience that has CGPointNull position.
```

Comment:
Is it correct to say that all "root layouts" should have a Null position?  And also that all non-root layouts should have a non-null position?

It would be helpful to clarify the use cases of these methods with changes to the comments, since otherwise it is tougher to get a sense of when these methods should be used.  For example, the conveniences that provide Null position are probably best used by ASDisplayNode subclasses that are manually creating a layout for -calculateLayoutThatFits:, or for ASLayoutSpec subclasses that are referencing the "self" level in the layout tree.